mne/epochs.py

```diff
@@ -912,6 +912,9 @@ def _compute_aggregate(self, picks, mode='mean'):
             if mode == "mean":
                 def fun(data):
                     return np.mean(data, axis=0)
+            elif mode == "median":
+                def fun(data):
+                    return np.median(data, axis=0)
             elif mode == "std":
                 def fun(data):
                     return np.std(data, axis=0)
```

Comment:
(In principle, the same appears here, though you'd have to use a lambda.)

Reply:
As `method` could also be a function, a try/catch or if/else it is at least necessary (plus the raise of an error if method is unknown). If we want to avoid a `if method in ('mean', 'median', 'std')`, we could ask for ~callable first, but the error must be inside the if as a try/catch or if/else. Also, `getattr` is not going to be fast. On the other hand, do the `if method in ('mean', 'median', 'std')` first, does not avoid the ifs itself.

So, perhaps, the best trade/off in this case is to enumerate all the cases.

Nevertheless, I prefer to avoid the lambda:
```
if mode in ('mean', 'median', 'std'):
    def fun(data):
        return getattr(np, mode)(data, axis=0)
else:
    ...
```
or maybe not:
```
fun = lambda data: getattr(np, mode)(data, axis=0) if method in ('mean', 'median', 'std') else mode
try:
    data = fun(self._data)
except ValueError:
    print(("mode must be mean, median, std, or callable"", got %s (type %s)." % (mode, type(mode))))
    raise
```
which one seems the best in this case?

Reply:
Oh it's fine as is I'd say... It's only a few lines, and they're fairly clear.

But I meant something like: `if method in ('median', 'mean', 'std'): fun = lambda x: getattr(np, method)(x, axis=0)`

Reply:
But we shouldn't try to be overly clever, it was just a note :)

Reply:
Ok! :) In any case, I can make the changes if necessary.